### ISSUE TITLE
Consider cleaning up the use of reflection in FunctionRSocketMessageHandler

### ISSUE THREAD

**Issue by olegz** [spring-cloud/spring-cloud-function#650]
Perhaps we can have something like 

```
public final void registerHandler(Function function, T mapping)
```

Given that Function is a public interface with a well known method name

```
X:
I think Oleg had to do it because Spring Framework only really has support for reflective message handlers without going down to a very low level in RSocket API.

X:
So you have this:
	private static final Method FUNCTION_APPLY_METHOD =
		ReflectionUtils.findMethod(Function.class, "apply", (Class<?>[]) null);

X:
It's a well known method on a JDK interface. There ought to be a way to just call Function.apply() I would have hoped.

X:
It seems like there is no equivalent of Webflux.fn or MVC.fn for RSockets (or other @MessageMappings).

X:
Maybe that's a sign of the same limitation?

Y:
No, we don't have a functional API. There is an issue for it:
https://github.com/spring-projects/spring-framework/issues/23135 (edited) 

X:
Thanks for the link

Z:
I think it would be nice to just have something like public final void registerHandler(Function function, T mapping) . Basically, Function being a well known interface, there is nothing to discover there other then always calling apply

X:
Except that type erasure would make it hard to do content negotation

X:
But, something like that.

X:
Spring Cloud Function reduces everything to a Function<Message<byte[]>,Message<byte[]>> or something?

Z:
actually that is not a problem since no conversion needs to happen until the actual function is invoked and it’s already wrapped with appropriate converter infrastructure, so Function<?, ?> is fine

Z:
reduces everything  - not really

Z:
it’s the MessageConverters that are there now have that expectation

Z:
but nothing should stop us to add additional message converters and we already allow users to add their own

Z:
basically what I am saying is that during the time when conversion needs to happen the in/out types are known.  .  . and that is and SCF level, not at  RSocket invocation

X:
You could make that class public in SCF I guess

X:
and add the method there

Z:
what class?

X:
FunctionRSocketMessageHandler

Z:
right

Z:
hmm, let me raise an issue and see what can be done at SCF level

X:
That wouldn't save you from having to use that reflection hack above though sadly

X:
But it might help me with my fake server spike
```

**Comment by olegz** [comment]
Also see https://github.com/spring-projects/spring-framework/issues/23135

**Comment by olegz** [comment]
Closing it as it would require many changes in RSocket's AbstractMethodMessageHandler
